commons-test/src/main/java/org/infinispan/commons/test/TestSuiteProgress.java

```diff
@@ -13,66 +13,75 @@
 
-   private static AtomicInteger failed = new AtomicInteger(0);
-   private static AtomicInteger succeeded = new AtomicInteger(0);
-   private static AtomicInteger skipped = new AtomicInteger(0);
+   private AtomicInteger failed = new AtomicInteger(0);
+   private AtomicInteger succeeded = new AtomicInteger(0);
+   private AtomicInteger skipped = new AtomicInteger(0);
+   private final PrintStream out;
+   private final boolean useColor;
+
+   public TestSuiteProgress() {
+      // Use a system property to avoid color
+      useColor = !Boolean.getBoolean("ansi.strip");
+      out = System.out;
+   }
 
-   static void testStarted(String name) {
+   void testStarted(String name) {
       String message = "Test starting: " + name;
-      consoleLog(message);
+      progress(message);
       log.info(message);
    }
 
-   static void testFinished(String name) {
+   void testFinished(String name) {
+      succeeded.incrementAndGet();
       String message = "Test succeeded: " + name;
-      consoleLog(message);
+      progress(message, GREEN);
       log.info(message);
-      succeeded.incrementAndGet();
-      printStatus();
+
    }
 
-   static void testFailed(String name, Throwable exception) {
+   void testFailed(String name, Throwable exception) {
+      failed.incrementAndGet();
       String message = "Test failed: " + name;
-      consoleLog(message);
+      progress(message, RED);
       log.error(message, exception);
-      failed.incrementAndGet();
-      printStatus();
    }
 
-   static void testIgnored(String name) {
+   void testIgnored(String name) {
+      skipped.incrementAndGet();
       String message = "Test ignored: " + name;
-      consoleLog(message);
+      progress(message, YELLOW);
       log.info(message);
-      skipped.incrementAndGet();
-      printStatus();
    }
 
-   static void testAssumptionFailed(String name, Throwable exception) {
+   void testAssumptionFailed(String name, Throwable exception) {
+      skipped.incrementAndGet();
       String message = "Test assumption failed: " + name;
-      consoleLog(message);
+      progress(message, YELLOW);
       log.info(message, exception);
-      skipped.incrementAndGet();
-      printStatus();
    }
 
-   static void setupFailed(String name, Throwable exception) {
+   void setupFailed(String name, Throwable exception) {
+      failed.incrementAndGet();
       String message = "Test setup failed: " + name;
-      consoleLog(message);
+      progress(message, RED);
       log.error(message, exception);
-      failed.incrementAndGet();
-      printStatus();
    }
 
-   private static void printStatus() {
-      String message = "Tests succeeded: " + succeeded.get() + ", failed: " + failed.get() + ", skipped: " +
-            skipped.get();
-      consoleLog(message);
+   synchronized void progress(CharSequence message) {
+      out.printf("[OK: %5s, KO: %5s, SKIP: %5s] %s\n", succeeded.get(), failed.get(), skipped.get(), message);
```

Comment:
Could not test this on Windows to see if it matters, but was thinking maybe we should use ```%n``` instead of ```\n``` for maximum crossplatform compat.